### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by n1zea144**
A spring batch process similar to sample-publisher that queries the Request Tracker web service and publishes request/sample status on a NATS topic defined in application.properties.

Note - checkstyle is passing locally.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+  <properties>
+    <java.version>1.8</java.version>
+    <maven.compiler.version>1.8</maven.compiler.version>
+    <spring.boot.version>2.3.3.RELEASE</spring.boot.version>
+    <spring.version>5.2.8.RELEASE</spring.version>
+    <jackson.version>2.11.2</jackson.version>
+    <!-- metadb messaging -->
+    <cmo_metadb_messaging_java.group>com.github.mskcc</cmo_metadb_messaging_java.group>
+    <cmo_metadb_messaging_java.version>0cc7bcc0cafe00421acc79171b6bf2f03660d854</cmo_metadb_messaging_java.version>
```
Comment (ao508) [inline pom.xml:26]: updated the messaging library: 

new commit hash: `38bb705c25ff8380d78df7a45698d3f84086de17`

and new property to add 
```
nats.filter_subject=METADB.*
```

**Review by ao508 — COMMENTED** [review]
Incomplete review, still want to build and test myself

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-parent</artifactId>
+    <version>2.3.3.RELEASE</version>
+  </parent>
+
+  <properties>
+    <java.version>1.8</java.version>
+    <maven.compiler.version>1.8</maven.compiler.version>
+    <spring.boot.version>2.3.3.RELEASE</spring.boot.version>
+    <spring.version>5.2.8.RELEASE</spring.version>
```
Comment (ao508) [inline pom.xml:22]: yeah I think for now let's just revert to 5.2.6.RELEASE to be consistent with the other projects and then we can upgrade at a later time

**Review by ao508 — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+version: 2.1
+
+orbs:
+  maven: circleci/maven@1.0.3
+
+jobs:
+  run_checkstyle:
+    <<: *defaults
+    working_directory: /tmp/repos/cmo-metadb-sample-publisher
```
[the remark this answers is not in the document]
Comment (n1zea144) [reply .circleci/config.yml]: Good catch, done!

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (n1zea144) [reply pom.xml:22]: I think I grabbed 5.2.8 from the sample-publisher pom.  Let me know if you want to downgrade it.

File: pom.xml (hunk 1 above)
Comment (n1zea144) [reply pom.xml]: Done and done (I have to check if the app-props file in configuration has this property).

File: pom.xml (hunk 2 above)
Comment (n1zea144) [reply pom.xml:22]: Ok, I'll downgrade.

Comment (n1zea144) [reply pom.xml]: Done.

**Review by ao508 — APPROVED** [review]

**Pull request merged**